<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to magrene • magrene</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to magrene">
<meta property="og:description" content="magrene">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">magrene</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/magrene.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/almeidasilvaf/magrene/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to magrene</h1>
                        <h4 data-toc-skip class="author">Fabrício Almeida-Silva</h4>
            <address class="author_afil">
      Ghent University / VIB Center for Plant Systems Biology, Ghent, Belgium<br><h4 data-toc-skip class="author">Yves Van de Peer</h4>
            <address class="author_afil">
      Ghent University / VIB Center for Plant Systems Biology, Ghent, Belgium<br><h4 data-toc-skip class="date">2022-07-28</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/magrene/blob/HEAD/vignettes/magrene.Rmd" class="external-link"><code>vignettes/magrene.Rmd</code></a></small>
      <div class="hidden name"><code>magrene.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Network motifs are the building blocks of complex networks, and they can be interpreted as small genetic circuits. Identifying and counting motifs in gene regulatory networks can reveal important aspects of the evolution of transcriptional regulation. In particular, they can be used to explore the impact of gene duplication in the rewiring of regulatory interactions <span class="citation">(Mottes et al. 2021)</span>. <code>magrene</code> aims to identify and analyze motifs in (duplicated) gene regulatory networks to better comprehend the role of gene duplication in network evolution. The figure below shows the networks motifs users can identify with <code>magrene</code>.</p>
<div class="figure" style="text-align: center">
<img src="motifs_vignette.png" alt="Network motifs and functions to identify them. Shaded boxes indicate paralogs. Regulators and targets are indicated in purple and green, respectively."><p class="caption">
Network motifs and functions to identify them. Shaded boxes indicate paralogs. Regulators and targets are indicated in purple and green, respectively.
</p>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install <code>magrene</code> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"magrene"</span><span class="op">)</span></span></code></pre></div>
<p>Then, you can load the package with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/almeidasilvaf/magrene" class="external-link">magrene</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-description">Data description<a class="anchor" aria-label="anchor" href="#data-description"></a>
</h2>
<p>For this vignette, we will use three example data sets:</p>
<ol style="list-style-type: decimal">
<li><p><strong>gma_grn:</strong> A gene regulatory network inferred with <code>BioNERO</code> <span class="citation">(Almeida-Silva and Venancio 2022)</span> using soybean RNA-seq data from the Soybean Expression Atlas <span class="citation">(Machado et al. 2020)</span>.</p></li>
<li><p><strong>gma_ppi:</strong> A protein-protein interaction (PPI) network for soybean obtained from the STRING database <span class="citation">(Szklarczyk et al. 2021)</span>, filtered to contain only physical interactions with confidence score &gt; 0.4.</p></li>
<li><p><strong>gma_paralogs:</strong> Soybean paralogous gene pairs derived by whole-genome, tandem, proximal, transposed, and dispersed duplications (WGD, TD, PD, TRD, and DD, respectively). This data set was obtained from <span class="citation">Almeida-Silva et al. (2020)</span>.</p></li>
</ol>
<p>Networks are represented as edge lists. Let’s take a look at the data.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gma_grn</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gma_grn</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  Node1           Node2     Weight</span></span>
<span><span class="co">#&gt; 615572 Glyma.20G042000 Glyma.13G027300 0.09219597</span></span>
<span><span class="co">#&gt; 177435 Glyma.06G225200 Glyma.04G205700 0.09054177</span></span>
<span><span class="co">#&gt; 208602 Glyma.09G278100 Glyma.05G084800 0.08846030</span></span>
<span><span class="co">#&gt; 190115 Glyma.18G039200 Glyma.05G002100 0.08382400</span></span>
<span><span class="co">#&gt; 78998  Glyma.01G075800 Glyma.02G221200 0.08271081</span></span>
<span><span class="co">#&gt; 870271 Glyma.09G231700 Glyma.18G220700 0.08208945</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gma_ppi</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gma_ppi</span><span class="op">)</span></span>
<span><span class="co">#&gt;                node1           node2</span></span>
<span><span class="co">#&gt; 4971 Glyma.19G213200 Glyma.01G004300</span></span>
<span><span class="co">#&gt; 4985 Glyma.14G061100 Glyma.01G004300</span></span>
<span><span class="co">#&gt; 4995 Glyma.03G179000 Glyma.01G004300</span></span>
<span><span class="co">#&gt; 5021 Glyma.03G097900 Glyma.01G004300</span></span>
<span><span class="co">#&gt; 5037 Glyma.07G047700 Glyma.01G004300</span></span>
<span><span class="co">#&gt; 5053 Glyma.19G051900 Glyma.01G004300</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gma_paralogs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gma_paralogs</span><span class="op">)</span></span>
<span><span class="co">#&gt;        duplicate1      duplicate2 type</span></span>
<span><span class="co">#&gt; 1 Glyma.01G007200 Glyma.01G007300   TD</span></span>
<span><span class="co">#&gt; 2 Glyma.01G007400 Glyma.01G007500   TD</span></span>
<span><span class="co">#&gt; 3 Glyma.01G011500 Glyma.01G011600   TD</span></span>
<span><span class="co">#&gt; 4 Glyma.01G012100 Glyma.01G012200   TD</span></span>
<span><span class="co">#&gt; 5 Glyma.01G015200 Glyma.01G015300   TD</span></span>
<span><span class="co">#&gt; 6 Glyma.01G020600 Glyma.01G020700   TD</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="finding-motifs">Finding motifs<a class="anchor" aria-label="anchor" href="#finding-motifs"></a>
</h2>
<p>Motifs can be found using <code>find_*</code> motifs, as shown in Figure 1. Each function returns a character vector of motifs, and each motif has its own character representation. Let’s demonstrate how they work. For the sake of demonstration, we will only use WGD-derived paralogs. For GRN motifs, we will only use a smaller subset of the edge list.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Include only WGD-derived paralogs</span></span>
<span><span class="va">paralogs</span> <span class="op">&lt;-</span> <span class="va">gma_paralogs</span><span class="op">[</span><span class="va">gma_paralogs</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"WGD"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Keep only the top 30k edges of the GRN, remove "Weight" variable</span></span>
<span><span class="va">grn</span> <span class="op">&lt;-</span> <span class="va">gma_grn</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30000</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<div class="section level3">
<h3 id="ppi-v">PPI V<a class="anchor" aria-label="anchor" href="#ppi-v"></a>
</h3>
<p>PPI V motifs are paralogous proteins that share an interaction partner. To find them, you will use <code><a href="../reference/find_ppi_v.html">find_ppi_v()</a></code>. The character representation of PPI V motifs is:</p>
<p><span class="math display">\[
\text{paralog1-partner-paralog2}
\]</span></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find PPI V motifs</span></span>
<span><span class="va">ppi_v</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_ppi_v.html">find_ppi_v</a></span><span class="op">(</span><span class="va">gma_ppi</span>, <span class="va">paralogs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ppi_v</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Glyma.19G213200-Glyma.01G004300-Glyma.03G216600"</span></span>
<span><span class="co">#&gt; [2] "Glyma.07G034300-Glyma.01G004300-Glyma.08G207800"</span></span>
<span><span class="co">#&gt; [3] "Glyma.14G101700-Glyma.01G004300-Glyma.06G061100"</span></span>
<span><span class="co">#&gt; [4] "Glyma.12G028500-Glyma.01G004300-Glyma.11G103700"</span></span>
<span><span class="co">#&gt; [5] "Glyma.19G189600-Glyma.01G004300-Glyma.03G189300"</span></span>
<span><span class="co">#&gt; [6] "Glyma.15G006700-Glyma.01G004300-Glyma.07G029200"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="v">V<a class="anchor" aria-label="anchor" href="#v"></a>
</h3>
<p>V motifs are paralogous regulators that regulate the same target. These motifs can be created when a regulator undergoes a small-scale duplication. To find them, you will use <code><a href="../reference/find_v.html">find_v()</a></code>. The character representation of V motifs is:</p>
<p><span class="math display">\[
\text{regulator-&gt;target&lt;-regulator}
\]</span></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find V motifs</span></span>
<span><span class="va">v</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_v.html">find_v</a></span><span class="op">(</span><span class="va">grn</span>, <span class="va">paralogs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Glyma.01G177200&lt;-Glyma.16G128800-&gt;Glyma.02G058700"</span></span>
<span><span class="co">#&gt; [2] "Glyma.01G177200&lt;-Glyma.08G362400-&gt;Glyma.02G058700"</span></span>
<span><span class="co">#&gt; [3] "Glyma.01G177200&lt;-Glyma.02G103700-&gt;Glyma.02G058700"</span></span>
<span><span class="co">#&gt; [4] "Glyma.01G177200&lt;-Glyma.09G152400-&gt;Glyma.02G058700"</span></span>
<span><span class="co">#&gt; [5] "Glyma.01G177200&lt;-Glyma.20G151800-&gt;Glyma.02G058700"</span></span>
<span><span class="co">#&gt; [6] "Glyma.01G177200&lt;-Glyma.08G020300-&gt;Glyma.02G058700"</span></span></code></pre></div>
<p>As you can see, there are no V motifs in this example data set. When that is the case, <code>find_*</code> functions return NULL.</p>
</div>
<div class="section level3">
<h3 id="lambda">Lambda<a class="anchor" aria-label="anchor" href="#lambda"></a>
</h3>
<p>Lambda motifs are the opposite of V motifs: a single regulator that regulates two target genes that are paralogous. These motifs can be created when an ancestral target gene undergoes a small-scale duplication. To find them you will use <code><a href="../reference/find_lambda.html">find_lambda()</a></code>. The character representation of lambda motifs is:</p>
<p><span class="math display">\[
\text{target1&lt;-regulator-&gt;target2}
\]</span></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lambda</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_lambda.html">find_lambda</a></span><span class="op">(</span><span class="va">grn</span>, <span class="va">paralogs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lambda</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Glyma.01G132800&lt;-Glyma.07G126500-&gt;Glyma.03G035400"</span></span>
<span><span class="co">#&gt; [2] "Glyma.01G205300&lt;-Glyma.07G126500-&gt;Glyma.05G064600"</span></span>
<span><span class="co">#&gt; [3] "Glyma.01G205300&lt;-Glyma.08G227000-&gt;Glyma.05G064600"</span></span>
<span><span class="co">#&gt; [4] "Glyma.01G205300&lt;-Glyma.07G128700-&gt;Glyma.05G064600"</span></span>
<span><span class="co">#&gt; [5] "Glyma.01G205300&lt;-Glyma.07G087900-&gt;Glyma.05G064600"</span></span>
<span><span class="co">#&gt; [6] "Glyma.01G205300&lt;-Glyma.19G022200-&gt;Glyma.05G064600"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="delta">Delta<a class="anchor" aria-label="anchor" href="#delta"></a>
</h3>
<p>Delta motifs are pretty similar to lambda motifs, but here we take protein-protein interactions between targets into account. Thus, they are represented by a regulator that regulates two targets that interact at the protein level. They can be created by the same evolutionary mechanism of lambda motifs. To find them, you will use <code><a href="../reference/find_delta.html">find_delta()</a></code>. The character representation of delta motifs is:</p>
<p><span class="math display">\[
\text{target1&lt;-regulator-&gt;target2}
\]</span></p>
<p>To find delta motifs, you have two options:</p>
<ol style="list-style-type: decimal">
<li><p>Pass PPI edge list + a vector of previously identified lambda motifs (recommended).</p></li>
<li><p>Pass PPI edge list + GRN edge list + paralogs data frame. In this option, <code><a href="../reference/find_delta.html">find_delta()</a></code> will find lambda motifs first, then use the lambda vector to find delta motifs. If you have identified lambda motifs beforehand, it is way faster to pass the lambda vector to <code><a href="../reference/find_delta.html">find_delta()</a></code>, so you don’t have to do double work.</p></li>
</ol>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find delta motifs from lambda motifs</span></span>
<span><span class="va">delta</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_delta.html">find_delta</a></span><span class="op">(</span>edgelist_ppi <span class="op">=</span> <span class="va">gma_ppi</span>, lambda_vec <span class="op">=</span> <span class="va">lambda</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">delta</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Glyma.10G242600&lt;-Glyma.04G221800-&gt;Glyma.20G151500"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="bifan">Bifan<a class="anchor" aria-label="anchor" href="#bifan"></a>
</h3>
<p>Bifan motifs are the most complex: they are represented by two paralogous regulators that regulate the same set of two paralogous targets. They can be created when both the ancestral regulator and the ancestral target are duplicated by small-scale duplications, or when the genome undergoes a whole-genome duplication event. To find these motifs, you will use <code><a href="../reference/find_bifan.html">find_bifan()</a></code>. The character representation of bifan motifs is:</p>
<p><span class="math display">\[
\text{regulator1,regulator2-&gt;target1,target2}
\]</span></p>
<p>Under the hood, what <code><a href="../reference/find_bifan.html">find_bifan()</a></code> does it to find lambda motifs involving the same targets and check if their regulators are paralogs. Thus, if you have identified lambda motifs beforehand, it is much faster to simply give them to <code><a href="../reference/find_bifan.html">find_bifan()</a></code>, so it doesn’t have to find them again.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find bifans from lambda motifs</span></span>
<span><span class="va">bifan</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/find_bifan.html">find_bifan</a></span><span class="op">(</span>paralogs <span class="op">=</span> <span class="va">paralogs</span>, lambda_vec <span class="op">=</span> <span class="va">lambda</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">bifan</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Glyma.01G177200,Glyma.02G058700-&gt;Glyma.09G152400,Glyma.20G151800"</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="counting-motifs-and-evaluating-significance">Counting motifs and evaluating significance<a class="anchor" aria-label="anchor" href="#counting-motifs-and-evaluating-significance"></a>
</h2>
<p>As motifs are simple character vectors, one can count their frequencies with the base R <code><a href="https://rdrr.io/r/base/length.html" class="external-link">length()</a></code> function. For example, let’s count the frequency of each motif in our example data set:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    Motif <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PPI V"</span>, <span class="st">"V"</span>, <span class="st">"Lambda"</span>, <span class="st">"Delta"</span>, <span class="st">"Bifan"</span><span class="op">)</span>,</span>
<span>    Count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ppi_v</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">lambda</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">delta</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">bifan</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">count_df</span></span>
<span><span class="co">#&gt;    Motif Count</span></span>
<span><span class="co">#&gt; 1  PPI V  1521</span></span>
<span><span class="co">#&gt; 2      V   112</span></span>
<span><span class="co">#&gt; 3 Lambda   163</span></span>
<span><span class="co">#&gt; 4  Delta     1</span></span>
<span><span class="co">#&gt; 5  Bifan     1</span></span></code></pre></div>
<p>However, unless you have another data set to which you can compare your frequencies, counting is not enough. You need to evaluate the significance of your motif frequencies. One way to do that is by comparing your observed frequencies to a null distribution generated by counting motifs in <em>N</em> (e.g., 1000) simulated networks.<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> <code>magrene</code> allows you to generate null distributions of motif frequencies for each motif type with the function <code><a href="../reference/generate_nulls.html">generate_nulls()</a></code>. As generating the null distributions takes a bit of time, we will load pre-computed null distributions of <em>N</em> = 100. As a rule of thumb, you would probably want <em>N</em> &gt;= 1000.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load null distros</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">nulls</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">nulls</span><span class="op">)</span></span>
<span><span class="co">#&gt; $lambda</span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;      4      3     11     10      8     11     13     13     13     12      6 </span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;     16      8     11      8     11      8      5     10      8      6     11 </span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;      9      6     11     12      9      6      8      9     15      6     11 </span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;      5     13     16      8     13      7     10      7     10     22     18 </span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;      8      4     12     10     14      6     15     13      8     14      5 </span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;      7     10      8      6     11     14      9     10     14     15     13 </span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;     10     10      8      8      9     11     13     10      7      7     11 </span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;     11      7     11     13      8      4     11     11     14     11     18 </span></span>
<span><span class="co">#&gt; lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda lambda </span></span>
<span><span class="co">#&gt;     10     14     10     11      8      7     11      6     10     18      6 </span></span>
<span><span class="co">#&gt; lambda </span></span>
<span><span class="co">#&gt;      8 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $delta</span></span>
<span><span class="co">#&gt; delta delta delta delta delta delta delta delta delta delta delta delta delta </span></span>
<span><span class="co">#&gt;     0     0     2     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; delta delta delta delta delta delta delta delta delta delta delta delta delta </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; delta delta delta delta delta delta delta delta delta delta delta delta delta </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; delta delta delta delta delta delta delta delta delta delta delta delta delta </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; delta delta delta delta delta delta delta delta delta delta delta delta delta </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; delta delta delta delta delta delta delta delta delta delta delta delta delta </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     1 </span></span>
<span><span class="co">#&gt; delta delta delta delta delta delta delta delta delta delta delta delta delta </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; delta delta delta delta delta delta delta delta delta </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     1     0     0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $V</span></span>
<span><span class="co">#&gt; V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V </span></span>
<span><span class="co">#&gt; 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 </span></span>
<span><span class="co">#&gt; V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V </span></span>
<span><span class="co">#&gt; 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 </span></span>
<span><span class="co">#&gt; V V V V V V V V V V V V V V V V V V V V </span></span>
<span><span class="co">#&gt; 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $bifan</span></span>
<span><span class="co">#&gt; bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan bifan </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0     0     0     0     0 </span></span>
<span><span class="co">#&gt; bifan bifan bifan bifan bifan bifan bifan bifan bifan </span></span>
<span><span class="co">#&gt;     0     0     0     0     0     0     0     0     0</span></span></code></pre></div>
<p>As you can see, the output of <code><a href="../reference/generate_nulls.html">generate_nulls()</a></code> is a list of numeric vectors with the frequency of each motif type in the simulated networks<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. You can use the null distribution to calculate Z-scores for your observed frequencies, which are defined as below:</p>
<p><span class="math display">\[
Z = \frac{ n_{observed} - \bar{n}_{null} }{ \sigma_{null} }
\]</span></p>
<p>To calculate Z-scores, you can use the function <code><a href="../reference/calculate_Z.html">calculate_Z()</a></code>. As input, you need to give a list of observed frequencies and a list of nulls.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create list of observed frequencies</span></span>
<span><span class="va">observed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    lambda <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">lambda</span><span class="op">)</span>, </span>
<span>    bifan <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">bifan</span><span class="op">)</span>, </span>
<span>    V <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span>,</span>
<span>    delta <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">delta</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/calculate_Z.html">calculate_Z</a></span><span class="op">(</span><span class="va">observed</span>, <span class="va">nulls</span><span class="op">)</span></span>
<span><span class="co">#&gt;    lambda     delta         V     bifan </span></span>
<span><span class="co">#&gt; 44.064857  3.952596       Inf       Inf</span></span></code></pre></div>
<p>Now that you have Z-scores, you can use a cut-off of your choice to define significance.</p>
</div>
<div class="section level2">
<h2 id="evaluting-interaction-similarity">Evaluting interaction similarity<a class="anchor" aria-label="anchor" href="#evaluting-interaction-similarity"></a>
</h2>
<p>Finally, another interesting pattern you may want to analyze is the interaction similarity between paralogous gene pairs. Previous studies have demonstrated that the Sorensen-Dice similarity is a suitable index for interaction similarity <span class="citation">(Defoort, Van de Peer, and Carretero-Paulet 2019; Mottes et al. 2021)</span>, which is defined as:</p>
<p><span class="math display">\[
S(A,B) = \frac{2 \left| A \cap B \right|}{ \left|A \right| + \left| B \right|}
\]</span></p>
<p>where A and B are the interacting partners of nodes A and B. To calculate the Sorensen-Dice similarity for paralogous gene pairs, you can use the function <code><a href="../reference/sd_similarity.html">sd_similarity()</a></code>. Let’s demonstrate it by calculating the similarity between paralogs in the PPI network.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sim</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sd_similarity.html">sd_similarity</a></span><span class="op">(</span><span class="va">gma_ppi</span>, <span class="va">paralogs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sim</span><span class="op">)</span></span>
<span><span class="co">#&gt;            duplicate1      duplicate2 sorensen_dice</span></span>
<span><span class="co">#&gt; 11631 Glyma.01G004300 Glyma.05G223800             1</span></span>
<span><span class="co">#&gt; 11696 Glyma.01G205300 Glyma.05G064600             1</span></span>
<span><span class="co">#&gt; 11706 Glyma.01G207700 Glyma.05G061400             1</span></span>
<span><span class="co">#&gt; 12157 Glyma.01G004300 Glyma.08G030900             1</span></span>
<span><span class="co">#&gt; 12803 Glyma.01G207700 Glyma.11G034700             1</span></span>
<span><span class="co">#&gt; 13298 Glyma.01G163300 Glyma.16G123100             1</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">sim</span><span class="op">$</span><span class="va">sorensen_dice</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">#&gt;  0.2200  1.0000  1.0000  0.9812  1.0000  1.0000</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This document was created under the following conditions:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">sessioninfo</span><span class="fu">::</span><span class="fu"><a href="https://r-lib.github.io/sessioninfo/reference/session_info.html" class="external-link">session_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.1.1 (2021-08-10)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 20.04.3 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2022-07-28</span></span>
<span><span class="co">#&gt;  pandoc   2.11.4 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package     </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version</span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  BiocManager    1.30.18 <span style="color: #555555;">2022-05-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  BiocParallel   1.26.2  <span style="color: #555555;">2021-08-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocStyle    * 2.20.2  <span style="color: #555555;">2021-06-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bookdown       0.27    <span style="color: #555555;">2022-06-14</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  bslib          0.4.0   <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  cachem         1.0.6   <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  cli            3.3.0   <span style="color: #555555;">2022-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  desc           1.4.1   <span style="color: #555555;">2022-03-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  digest         0.6.29  <span style="color: #555555;">2021-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  evaluate       0.15    <span style="color: #555555;">2022-02-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  fastmap        1.1.0   <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  fs             1.5.2   <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  highr          0.9     <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  htmltools      0.5.3   <span style="color: #555555;">2022-07-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib      0.1.4   <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite       1.8.0   <span style="color: #555555;">2022-02-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  knitr          1.39    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  magrene      * 0.99.0  <span style="color: #555555;">2022-07-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  magrittr       2.0.3   <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  memoise        2.0.1   <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown        2.0.6   <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  purrr          0.3.4   <span style="color: #555555;">2020-04-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  R6             2.5.1   <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  ragg           1.2.2   <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  rlang          1.0.4   <span style="color: #555555;">2022-07-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown      2.14    <span style="color: #555555;">2022-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot      2.0.3   <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  sass           0.4.2   <span style="color: #555555;">2022-07-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo    1.2.2   <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  stringi        1.7.8   <span style="color: #555555;">2022-07-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  stringr        1.4.0   <span style="color: #555555;">2019-02-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  systemfonts    1.0.4   <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  textshaping    0.3.6   <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  xfun           0.31    <span style="color: #555555;">2022-05-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt;  yaml           2.3.5   <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-almeida2020exploring" class="csl-entry">
Almeida-Silva, Fabricio, Kanhu C Moharana, Fabricio B Machado, and Thiago M Venancio. 2020. <span>“Exploring the Complexity of Soybean (Glycine Max) Transcriptional Regulation Using Global Gene Co-Expression Networks.”</span> <em>Planta</em> 252 (6): 1–12.
</div>
<div id="ref-almeida2022bionero" class="csl-entry">
Almeida-Silva, Fabricio, and Thiago M Venancio. 2022. <span>“BioNERO: An All-in-One r/Bioconductor Package for Comprehensive and Easy Biological Network Reconstruction.”</span> <em>Functional &amp; Integrative Genomics</em> 22 (1): 131–36.
</div>
<div id="ref-defoort2019evolution" class="csl-entry">
Defoort, Jonas, Yves Van de Peer, and Lorenzo Carretero-Paulet. 2019. <span>“The Evolution of Gene Duplicates in Angiosperms and the Impact of Protein–Protein Interactions and the Mechanism of Duplication.”</span> <em>Genome Biology and Evolution</em> 11 (8): 2292–2305.
</div>
<div id="ref-machado2020systematic" class="csl-entry">
Machado, Fabricio B, Kanhu C Moharana, Fabricio Almeida-Silva, Rajesh K Gazara, Francisnei Pedrosa-Silva, Fernanda S Coelho, Clı́cia Grativol, and Thiago M Venancio. 2020. <span>“Systematic Analysis of 1298 RNA-Seq Samples and Construction of a Comprehensive Soybean (Glycine Max) Expression Atlas.”</span> <em>The Plant Journal</em> 103 (5): 1894–1909.
</div>
<div id="ref-mottes2021impact" class="csl-entry">
Mottes, Francesco, Chiara Villa, Matteo Osella, and Michele Caselle. 2021. <span>“The Impact of Whole Genome Duplications on the Human Gene Regulatory Networks.”</span> <em>PLoS Computational Biology</em> 17 (12): e1009638.
</div>
<div id="ref-szklarczyk2021string" class="csl-entry">
Szklarczyk, Damian, Annika L Gable, Katerina C Nastou, David Lyon, Rebecca Kirsch, Sampo Pyysalo, Nadezhda T Doncheva, et al. 2021. <span>“The STRING Database in 2021: Customizable Protein–Protein Networks, and Functional Characterization of User-Uploaded Gene/Measurement Sets.”</span> <em>Nucleic Acids Research</em> 49 (D1): D605–12.
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><strong>NOTE:</strong> Simulated networks are created by node label permutation (i.e., resampling node labels without replacement). This method allows you to have random networks that preserve the same degree of the original network. Hence, networks are called <strong>degree-preserving simulated networks</strong>.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><strong>Note on performance:</strong> The function <code><a href="../reference/generate_nulls.html">generate_nulls()</a></code> can be parallelized thanks to the Bioconductor package <em><a href="https://bioconductor.org/packages/3.13/BiocParallel" class="external-link">BiocParallel</a></em>. However, keep in mind that parallelization is not always the best choice, because it may take longer to create multiple copies of your data to split into multiple cores than it takes to find motifs with a single core.<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Fabrício Almeida-Silva, Yves Van de Peer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
